python/lsst/ap/association/packageAlerts.py

```diff
@@ -555,3 +575,15 @@ def _delivery_callback(self, err, msg):
             self.log.warning('Message failed delivery: %s\n' % err)
         else:
             self.log.debug('Message delivered to %s [%d] @ %d', msg.topic(), msg.partition(), msg.offset())
+
+    def _server_check(self):
+        try:
+            admin_client = AdminClient(self.kafkaAdminConfig)
+            topics = admin_client.list_topics(timeout=0.5).topics
+
+            if not topics:
+                raise RuntimeError()
+
+        except KafkaException as e:
+            self.log.error(e)
+            raise
```

Comment:
I may be missing some functionality of raise, does raise log the error automatically?

Reply:
All our execution frameworks should log any exceptions that get emitted from the pipeline (certainly Prompt Processing does!). It's generally considered bad practice to log an exception you're not handling, not least because of what happens if _everybody_ does it. 🙂

Reply:
I've swapped it to      
```   
except KafkaException as exception:
          raise exception
```
However, if I didn't include the `as expection` part and just did `raise` by itself, would that catch the full error correctly?

Reply:
It would, but I don't think you need that at all... you could remove that whole `try`-`except` block and the behavior would be the same. (It might be helpful to document that `KafkaException` is expected; that usually goes [in the docstring](https://developer.lsst.io/python/numpydoc.html#py-docstring-raises).)

Reply:
Ah, that is very true. I'll add a docstring (something I should have done anyways) and remove the try/except.